task/git-cli/0.2/README.md

```diff
@@ -0,0 +1,60 @@
+# Git Task
+
+This `Task` is Git task to work with repositories used by other tasks
+in your Pipeline.
+
+## `git-cli`
+
+This [task](../0.2/git-cli.yaml) can be used to perform `git operations`.
+All git commands can be found [here](https://git-scm.com/docs).
+
+Command that needs to be run can be passed as a script to the task.
+
+### Workspaces
+
+* **source**: A workspace that contains the fetched git repository.
+* **input**: A workspace that contains file that needs to be added to git.
+* **ssh-directory**: An optional workspace to provide SSH credentials. At
+  minimum this should include a private key but can also include other common
+  files from `.ssh` including `config` and `known_hosts`. It is **strongly**
+  recommended that this workspace be bound to a Kubernetes `Secret`.
+* **basic-auth**: An optional workspace containing `.gitconfig` and
+  `.git-credentials` files. This allows username/password/access token to be
+  provided for basic auth.
+
+  It is **strongly** recommended that this workspace be bound to a Kubernetes
+  `Secret`. For details on the correct format of the files in this Workspace
+  see the `git-clone` (0.4) documentation as this task is based on this.
+
+### Parameters
+
+* **BASE_IMAGE**: The base image for the task.
+(_default_: `alpine/git:latest`)
+* **GIT_USER_NAME**: Git user name for performing git operation.
+* **GIT_USER_EMAIL**:  Git user email for performing git operation.
+* **GIT_SCRIPT**: The git script to run. (_required_)
+* **VERBOSE**: Log the commands that are executed during `git-clone`'s operation. (_default_: true)
+* **USER_HOME**: The user's home directory. Set this explicitly if you are running the image as a non-root user. (_default_: "/tekton/home")
+
+### Results
+
+* **commit**: The precise commit SHA after git operation is performed.
+
+### Usage
+
+This task needs authentication to git in order to push after the git operation.
+See the `git-clone` documentation for more information.
```

Comment:
I think it might be better to duplicate the relevant sections of git-clone's doc.  That way a reader doesn't get bounced around between markdown files. Also git-clone's docs could get changed in future and references like this may become stale by accident.

Reply:
Added the SSH docs as well.